xgi/generators/uniform.py

```diff
@@ -89,3 +100,253 @@ def uniform_hypergraph_configuration_model(k, m, seed=None):
+        tensor of probabilities between communities
+    sizes : list or 1D numpy array
+        The sizes of the community blocks in order
+    seed : integer or None (default)
+        The seed for the random number generator
+
+    Returns
+    -------
+    Hypergraph
+        The constructed SBM hypergraph
+
+    Raises
+    ------
+    XGIError
+        - If the length of sizes and p do not match.
+        - If p is not a tensor with every dimension equal
+        - If p is not m-dimensional
+        - If the entries of p are not in the range [0, 1]
+        - If the sum of the vector of sizes does not equal the number of nodes.
+    Exception
+        If there is an integer overflow error
+    
+    See Also
+    --------
+    uniform_hypergraph_planted_partition_model
+
+    References
+    ----------
+    Nicholas W. Landry and Juan G. Restrepo.
+    "Polarization in hypergraphs with community structure."
+    Preprint, 2023. https://doi.org/10.48550/arXiv.2302.13967
+    """
+
+    # Check if dimensions match
+    if len(sizes) != np.size(p, axis=0):
+        raise XGIError("'sizes' and 'p' do not match.")
+    if len(np.shape(p)) != m:
+        raise XGIError("The dimension of p does not match m")
+    # Check that p has the same length over every dimension.
+    if len(set(np.shape(p))) != 1:
+        raise XGIError("'p' must be a square tensor.")
+    if np.max(p) > 1 or np.min(p) < 0:
+        raise XGIError("Entries of 'p' not in [0,1].")
+    if np.sum(sizes) != n:
+        raise XGIError("Sum of sizes does not match n")
+
+    if seed is not None:
+        random.seed(seed)
+
+    node_labels = range(n)
+    H = empty_hypergraph()
+    H.add_nodes_from(node_labels)
+
+    block_range = range(len(sizes))
+    # Split node labels in a partition (list of sets).
+    size_cumsum = [sum(sizes[0:x]) for x in range(0, len(sizes) + 1)]
+    partition = [
+        list(node_labels[size_cumsum[x] : size_cumsum[x + 1]])
+        for x in range(0, len(size_cumsum) - 1)
+    ]
+
+    for block in itertools.product(block_range, repeat=m):
+        if p[block] == 1:  # Test edges cases p_ij = 0 or 1
+            edges = itertools.product((partition[i] for i in block_range))
+            for e in edges:
+                H.add_edge(e)
+        elif p[block] > 0:
+            partition_sizes = [len(partition[i]) for i in block]
+            max_index = reduce(operator.mul, partition_sizes, 1)
+            if max_index < 0:
+                raise Exception("Index overflow error!")
+            index = np.random.geometric(p[block]) - 1
+
+            while index < max_index:
+                indices = _index_to_edge_partition(index, partition_sizes, m)
+                e = [partition[block[i]][indices[i]] for i in range(m)]
+                if len(e) == m:
+                    H.add_edge(e)
+                index += np.random.geometric(p[block])
+    return H
+
+
+def uniform_hypergraph_planted_partition_model(n, m, rho, k, epsilon):
```

Comment:
Naming not consistent with the SBM function above. Did you not use the acronym because less known than SBM?

Reply:
good point. I will write as an acronym and explain in the docstring.